packages/flutter/lib/src/painting/circle_border.dart

```diff
@@ -18,16 +18,27 @@ import 'edge_insets.dart';
 /// When applied to a rectangular space, the border paints in the center of the
 /// rectangle.
 ///
+/// The [eccentricity] parameter allows the circle to be painted touching all
+/// the edges of a rectangle, becoming an oval. When applied to a
+/// squared space, [eccentricity] is ignored.
+///
 /// See also:
 ///
+///  * [OvalBorder], which draws a Circle touching all the edges of the box.
 ///  * [BorderSide], which is used to describe each side of the box.
-///  * [Border], which, when used with [BoxDecoration], can also
-///    describe a circle.
+///  * [Border], which, when used with [BoxDecoration], can also describe a circle.
 class CircleBorder extends OutlinedBorder {
   /// Create a circle border.
   ///
   /// The [side] argument must not be null.
-  const CircleBorder({ super.side }) : assert(side != null);
+  const CircleBorder({ super.side, this.eccentricity = 0 })
+      : assert(side != null && eccentricity != null && eccentricity >= 0.0 && eccentricity <= 1.0);
```

Comment:
You might split these into separate asserts for readability, and so when they fire they are more useful.

You can also print the offending value in the assert message so it's more helpful. Like so:

```dart
assert(eccentricity >= 0.0, 'The eccentricity argument $eccentricity is not greater than or equal to zero.');
```

Reply:
Yeah, it does seem kind of silly, but there are people out there still running without strict null safety, which means that the asserts still do something.

We'll eventually make a pass and remove all such asserts with a tool once it's been long enough since we turned on null safety.

Reply:
I thought even when null-safety was off when calling code that was null-safe it would be null-safe. Ok, adding `eccentricity != null`